python/cuml/test/dask/test_comms.py

```diff
@@ -0,0 +1,87 @@
+# Copyright (c) 2019, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import pytest
+from dask_cuda import LocalCUDACluster
+
+from dask.distributed import Client, wait
+
+from cuml.dask.common import CommsBase
+from cuml.dask.common import perform_test_comms_send_recv
+from cuml.dask.common import perform_test_comms_allreduce
+
+
+def test_comms_init_no_p2p():
+
+    cluster = LocalCUDACluster(threads_per_worker=1)
+    client = Client(cluster)  # noqa
+
+    cb = CommsBase()
+    cb.init()
+
+    assert cb.nccl_initialized is True
+    assert cb.ucx_initialized is False
+
+
+def test_allreduce():
+
+    cluster = LocalCUDACluster(threads_per_worker=1)
+    client = Client(cluster)
+
+    cb = CommsBase()
+    cb.init()
+
+    workers = client.has_what().keys()
+
+    print(str(workers))
+
+    dfs = [client.submit(perform_test_comms_allreduce, handle, workers=[w])
+           for wid, w, handle in cb.handles]
+
+    wait(dfs)
+
+    print(str(list(map(lambda x: x.result(), dfs))))
+
+    assert all(list(map(lambda x: x.result(), dfs)))
+
+    # todo: Destroy is failing here. Need to fix it
```

Comment:
Do we have a github issue to look into this? Also this is not the reason for the CI logging warnings in CI, right? The ones that’s look like 

```
--- Logging error ---
Traceback (most recent call last):
  File "/conda/envs/gdf/lib/python3.6/logging/__init__.py", line 996, in emit
    stream.write(msg)
  File "/conda/envs/gdf/lib/python3.6/site-packages/_pytest/capture.py", line 427, in write
    self.buffer.write(obj)
ValueError: I/O operation on closed file
Call stack:
  File "/conda/envs/gdf/lib/python3.6/site-packages/distributed/process.py", line 335, in _cleanup_dangling
    logger.warning("reaping stray process %s" % (proc,))
Message: 'reaping stray process <ForkServerProcess(Dask Worker process (from Nanny), started daemon)>'
Arguments: ()
```

Those come from the dask tests, right?

Reply:
The issue referenced in this comment is UCX-ralated and is not a priority since UCX is experimental, optional, and turned off by default in 0.9.